src/Microsoft.Azure.SignalR.Common/Interfaces/IServiceEndpointProvider.cs

```diff
@@ -15,7 +15,7 @@ internal interface IServiceEndpointProvider
 
         string GetClientEndpoint(string hubName, string originalPath, string queryString);
 
-        Task<string> GenerateServerAccessTokenAsync(string hubName, string userId, TimeSpan? lifetime = null);
+        IAccessTokenProvider GetServerAccessTokenProvider(string hubName, string serverId);
```

Comment:
why serverId is a parameter, it sounds like a static field per app

Reply:
It's by our current design. 

`ConnectionFactory` generated the `serverId` and make sure this value in the HTTP header and in the identifier claim are equal.